enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -771,3 +775,35 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 		}
 	}
 }
+
+func (s *ExecutionServer) updateRouter(ctx context.Context, taskID string, executeResponse *repb.ExecuteResponse) error {
+	router := s.env.GetTaskRouter()
+	if router == nil {
+		return nil
+	}
+	// If the result was served from cache, nothing was actually executed, so no
+	// need to update the task router.
+	if executeResponse.GetCachedResult() {
+		return nil
+	}
+	// TODO: Maybe pre-fetch the command at the beginning of PublishOperation,
```

Comment:
nit: can you remove this comment? (It's kinda 'speculative')

Reply:
Done